llama_stack/models/llama/datatypes.py

```diff
@@ -47,7 +47,9 @@ class BuiltinTool(Enum):
 class ToolCall(BaseModel):
     call_id: str
     tool_name: Union[BuiltinTool, str]
-    arguments: Dict[str, RecursiveType]
+    arguments: Union[str, Dict[str, RecursiveType]]
```

Comment:
can you add a comment here explaining why there is a `Union[str, ...]` and how the deprecation process will happen?